### ISSUE TITLE
Container won't start after Proxmox Host reboot (what's mp0 for?)

### ISSUE THREAD

**Issue by sodaws** [DazzyWalkman/oplxc4pve#2]
Hi,
I recently stumbled on lxd-openwrt-simplified and this project and first of all, I want to thank you for your work. You saved me from a lot of headaches.

I have a problem that might be more related to proxmox than your script but I thought since you spent quite some time with Proxmox and OpenWRT containers, you might have solved this problem before.

Basically, if I create a container with the script, start the container, reboot the PVE-Host and then try to start the container again, it fails to do so.

Here are the logs:

```
root@Proxmox:~# lxc-start -n 201 -l debug -o /tmp/201.log
lxc-start: 201: lxccontainer.c: wait_on_daemonized_start: 841 No such file or directory - Failed to receive the container state
lxc-start: 201: tools/lxc_start.c: main: 308 The container failed to start
lxc-start: 201: tools/lxc_start.c: main: 311 To get more details, run the container in foreground mode
lxc-start: 201: tools/lxc_start.c: main: 314 Additional information can be obtained by setting the --logfile and --logpriority options
root@Proxmox:/tmp# cat 201.log 
lxc-start 201 20210405162532.214 INFO     confile - confile.c:set_config_idmaps:2055 - Read uid map: type u nsid 0 hostid 100000 range 65536
lxc-start 201 20210405162532.215 INFO     confile - confile.c:set_config_idmaps:2055 - Read uid map: type g nsid 0 hostid 100000 range 65536
lxc-start 201 20210405162532.216 INFO     lxccontainer - lxccontainer.c:do_lxcapi_start:969 - Set process title to [lxc monitor] /var/lib/lxc 201
lxc-start 201 20210405162532.216 DEBUG    lxccontainer - lxccontainer.c:wait_on_daemonized_start:830 - First child 30443 exited
lxc-start 201 20210405162532.217 INFO     lsm - lsm/lsm.c:lsm_init:29 - LSM security driver AppArmor
lxc-start 201 20210405162532.217 INFO     conf - conf.c:run_script_argv:340 - Executing script "/usr/share/lxc/hooks/lxc-pve-prestart-hook" for container "201", config section "lxc"
lxc-start 201 20210405162533.213 DEBUG    conf - conf.c:run_buffer:312 - Script exec /usr/share/lxc/hooks/lxc-pve-prestart-hook 201 lxc pre-start produced output: directory '/run/ctshare/' does not exist

lxc-start 201 20210405162533.231 ERROR    conf - conf.c:run_buffer:323 - Script exited with status 2
lxc-start 201 20210405162533.231 ERROR    start - start.c:lxc_init:797 - Failed to run lxc.hook.pre-start for container "201"
lxc-start 201 20210405162533.231 ERROR    start - start.c:__lxc_start:1896 - Failed to initialize container "201"
lxc-start 201 20210405162533.231 INFO     conf - conf.c:run_script_argv:340 - Executing script "/usr/share/lxcfs/lxc.reboot.hook" for container "201", config section "lxc"
lxc-start 201 20210405162533.735 INFO     conf - conf.c:run_script_argv:340 - Executing script "/usr/share/lxc/hooks/lxc-pve-poststop-hook" for container "201", config section "lxc"
lxc-start 201 20210405162534.708 DEBUG    conf - conf.c:run_buffer:312 - Script exec /usr/share/lxc/hooks/lxc-pve-poststop-hook 201 lxc post-stop produced output: umount: /var/lib/lxc/.pve-staged-mounts/mp0: not mounted.

lxc-start 201 20210405162534.708 DEBUG    conf - conf.c:run_buffer:312 - Script exec /usr/share/lxc/hooks/lxc-pve-poststop-hook 201 lxc post-stop produced output: command 'umount -- /var/lib/lxc/.pve-staged-mounts/mp0' failed: exit code 32

lxc-start 201 20210405162534.770 ERROR    lxccontainer - lxccontainer.c:wait_on_daemonized_start:841 - No such file or directory - Failed to receive the container state
lxc-start 201 20210405162534.770 ERROR    lxc_start - tools/lxc_start.c:main:308 - The container failed to start
lxc-start 201 20210405162534.770 ERROR    lxc_start - tools/lxc_start.c:main:311 - To get more details, run the container in foreground mode
lxc-start 201 20210405162534.771 ERROR    lxc_start - tools/lxc_start.c:main:314 - Additional information can be obtained by setting the --logfile and --logpriority options
root@Proxmox:~# pct mount 201
mounting container failed
directory '/run/ctshare/' does not exist
```

There seems to be a problem with the mountpoint mp0. If I go into /etc/pve/lxc/201.conf and comment out the line `mp0%3A /run/ctshare/,mp=/shared` , the container can start again.

I'm a noob with Proxmox and OpenWRT. What does the mountpoint do or rather what is it for? Do I need it or can I just leave it like this?

**Comment by DazzyWalkman** [comment]
This is a bind mount directory for file sharing between host and ct guests,  which is needed when this oplxc4pve script performing ct upgrade operation, where the old ct dumps its config backup.
Currently, the dir is /run/ctshare, which will get removed upon host reboot. And its recreation is handled by the hookscript (pls see files/hookscript/oplxc.pl for detail ) attatched to the ct. 
As hookscripts may vary for different user cases, I leave it to the users to make their own hookscripts. 
So the users have to do some minor configuration including nics and hookscript upon creating first ct of the oplxc series.  The settings will be copied over automatically by the oplxc4pve during upgrade operation afterwards.
Users can modify the script, to point this host mp path to a persistent directory, or use pct push/pull instead. 
The mp thing suits me well as I have some additional tasks like scheduled log dumping, where pct push/pull is too much for my taste.

**Comment by DazzyWalkman** [comment]
Close with fe2a4884f8028922fabdad6cbd77933df1ba4bde

**Comment by DazzyWalkman** [comment]
Fix further with c50b4754cd6dddc43a6d4912706a3bc4a38a9c45
